features/netsocket/nsapi_types.h

```diff
@@ -254,6 +254,7 @@ typedef enum nsapi_socket_option {
     NSAPI_RCVBUF,            /*!< Sets recv buffer size */
     NSAPI_ADD_MEMBERSHIP,    /*!< Add membership to multicast address */
     NSAPI_DROP_MEMBERSHIP,   /*!< Drop membership to multicast address */
+    NSAPI_MULTITTL,          /*!< Multicast TTL */
```

Comment:
This should have the full name `NSAPI_MULTICAST_TTL`. And if adding it, we'd want it done for Nanostack too. What's it got to do with the ODIN driver? Might be better in a separate PR.